### PR TITLE
Add builder to HtmxResponse

### PR DESCRIPTION
**Pull request by xhaggi**

### CODE REVIEW

**Comment by xhaggi** [comment]
There is one thing I would like to discuss. What do you think if we rename the `template` method in the builder to `view`, as you pass a view name (string), a View or a ModelAndView to this method? Spring also call it "view" in its documentation.

```java
HtmxResponse.builder().view("myview").build();
```

**Comment by wimdeblauwe** [comment]
Interesting thought, although it might be strange to have a `ModelAndView` as parameter to a `view(...)` method. @dsyer what do you think about it?

**Comment by dsyer** [comment]
I don't really care if it's "template" or "view". Changing "addTemplate" to "template" (and a bunch of other changes in this rather large PR) is already a breaking change though, so probably shouldn't be in a minor release? I'm OK with things the way they are but if you want to change then it would need to be version 4.0.0 right?

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java [hunk 1]
```diff
@@ -196,96 +249,321 @@ public HtmxResponse and(HtmxResponse otherResponse) {
...
+            return this;
+        }
+
+        /**
+         * Append the rendered template or fragment as a {@link ModelAndView}.
+         *
+         * @param template must not be {@literal null}.
+         * @return the builder
+         */
+        public Builder template(ModelAndView template) {
```
Comment (checketts) [inline htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java:391]: I recall there being talk about being able to add separate model that are rendered with their separate views. Is that now in HTMX? That would be cool and align with this. I would support adding the `view` call in addition to template.

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java [hunk 2]
```diff
@@ -196,96 +249,321 @@ public HtmxResponse and(HtmxResponse otherResponse) {
...
+    @Deprecated
     private boolean same(ModelAndView one, ModelAndView two) {
         if (one == two) {
             return true;
         }
         if (one == null || two == null) {
             return false;
         }
-        if (one.getViewName() != null && one.getViewName().equals(two.getViewName())) {
+        if (one.getViewName() !=null && one.getViewName().equals(two.getViewName())) {
```
Comment (wimdeblauwe) [inline htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java:279]: Please keep the space between `!=` and `null`

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java [hunk 3]
```diff
@@ -196,96 +249,321 @@ public HtmxResponse and(HtmxResponse otherResponse) {
...
             return true;
         }
-        if (one.getView() != null && one.getView().equals(two.getView())) {
+        if (one.getView() !=null && one.getView().equals(two.getView())) {
```
Comment (wimdeblauwe) [inline htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java:282]: Please keep the space between `!=` and `null`

**Comment by wimdeblauwe** [comment]
> I don't really care if it's "template" or "view". Changing "addTemplate" to "template" (and a bunch of other changes in this rather large PR) is already a breaking change though, so probably shouldn't be in a minor release? I'm OK with things the way they are but if you want to change then it would need to be version 4.0.0 right?

It is not breaking at this point, as the current methods are still there and marked as deprecated. @xhaggi If Spring also calls it view in their docs, then I am ok with changing it to `view()` in the builder.

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java (hunk 2 above)
Comment (xhaggi) [reply htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java]: Done.

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java (hunk 3 above)
Comment (xhaggi) [reply htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java]: Done.

**Comment by xhaggi** [comment]
Ok great. The builder now uses `view` and everything else is still the same. I also spent some time reworking the `HtmxResponseTest` class. Now all the tests for the deprecated methods have gone into a nested test class, so each test exists for both. In the course of this, I have also revised the naming of the test methods.

File: README.md [hunk 4]
```diff
@@ -132,9 +132,10 @@ as the return type of a controller method, where you can add multiple templates.
 @GetMapping("/partials/main-and-partial")
 public HtmxResponse getMainAndPartial(Model model){
     model.addAttribute("userCount", 5);
-    return new HtmxResponse()
-        .addTemplate("users-list")
-        .addTemplate("users-count");
+    return HtmxResponse.builder()
+        .template("users-list")
```
Comment (wimdeblauwe) [inline README.md:136]: These examples should also use the `view()` method.

Comment (xhaggi) [reply README.md]: Done.

**Pull request merged**